### DISCUSSION TITLE
Unrecognized argument running pytest with plugin from executable file

### DISCUSSION THREAD

**Discussion by sush1q** [pytest-dev/pytest#10186] [Q&A] [answered]
I have script like this to run my tests:
```
import pytest
import src
import ioctl_module
import allure
import allure_pytest

pytest.main(["-s", src.__path__[0], "--alluredir=./allure_results"])
```
While I am running it with IDE Debugger it perfectly works and generate allure-report
But, when I pack this script with PyInstaller I have this problem while running it:
```
ERROR: usage: run_tests [options] [file_or_dir] [file_or_dir] [...]
run_tests: error: unrecognized arguments: --alluredir=./allure_results
  inifile: None
  rootdir: /home/administrator/all_com
```

I used pyinstaller debugger with import info, there I saw that allure was imported
used pyinstaller command:

`pyinstaller run_tests.py  --noconfirm --onedir --name run_tests --add-data src --collect-submodules allure-pytest --collect-submodules allure -d imports`

**To Reproduce**
1. pack script by pyinstaller
2. run packed script

**Expected behavior**
I wanted to generate allure-report after runned tests packed with executable file

**Screenshots**
![image](https://user-images.githubusercontent.com/89254778/182857508-9e0517d3-3012-42b3-84a3-7725c7205b7c.png)

**Environment (please complete the following information):**

| Allure version | 2.9.0 |
| --- | --- |
| Test framework | pytest |
| Allure adaptor | allure-pytest@2.9.45 |
| Generate report using | allure-pytest@2.9.45 |

**Comment by nicoddemus** [comment, marked as answer]
Hi @sush1q,

 
Try this:

```python
...
import allure_pytest.plugin

pytest.main(["-s", src.__path__[0], "--alluredir=./allure_results"], plugins=[allure_pytest.plugin])
```

**Reply by nicoddemus** [reply]
Related: https://github.com/pyinstaller/pyinstaller/issues/305

**Reply by nicoddemus** [reply]
Explanation: pytest uses setuptools entry points to register plugins, and that mechanism is not supported by PyInstaller (not sure if the support has been added/is working in the latest versions).
